test/src/api/commands/client/testSetDeviceDimensions.js

```diff
@@ -0,0 +1,184 @@
+            browserName: 'chrome',
+            browserVersion: '92.0'
+          }
+        }
+      },
+      method: 'POST',
+      statusCode: 201
+    }, true);
+
+    Nightwatch.initW3CClient({
+      desiredCapabilities: {
+        browserName: 'chrome',
+        'goog:chromeOptions': {}
+      }
+    }).then(client => {
+
+      let expectedCDPCommand;
+      let expectedWidth;
+      let expectedHeight;
+      let expectedDeviceScaleFactor;
+      let expectedMobile;
+
+      client.transport.driver.createCDPConnection = function() {
+        return Promise.resolve({
+          execute: function(command, metrics) {
+            expectedCDPCommand = command;
+            expectedWidth = metrics.width;
+            expectedHeight = metrics.height;
+            expectedDeviceScaleFactor = metrics.deviceScaleFactor;
+            expectedMobile = metrics.mobile;
+          }
+        });
+      };
+      client.api.setDeviceDimensions({width: 400, height: 600, deviceScaleFactor: 50, mobile: true}, function (){
+        assert.strictEqual(expectedCDPCommand, 'Emulation.setDeviceMetricsOverride');
+        assert.strictEqual(expectedWidth, 400);
+        assert.strictEqual(expectedHeight, 600);
+        assert.strictEqual(expectedDeviceScaleFactor, 50);
+        assert.strictEqual(expectedMobile, true);
+      });
+      client.start(done);
+    });
+  });
+
+  // Should set device dimensions when just two properties are provided.
+  it('browser.setDeviceDimensions({width, height})', function (done) {
+
+    MockServer.addMock({
+      url: '/session',
+      response: {
+        value: {
+          sessionId: '13521-10219-202',
+          capabilities: {
+            browserName: 'chrome',
+            browserVersion: '92.0'
+          }
+        }
+      },
+      method: 'POST',
+      statusCode: 201
+    }, true);
+
+    Nightwatch.initW3CClient({
+      desiredCapabilities: {
+        browserName: 'chrome',
+        'goog:chromeOptions': {}
+      }
+    }).then(client => {
+
+      let expectedCDPCommand;
+      let expectedWidth;
+      let expectedHeight;
+      let expectedDeviceScaleFactor;
+      let expectedMobile;
+
+      client.transport.driver.createCDPConnection = function() {
+        return Promise.resolve({
+          execute: function(command, metrics) {
+            expectedCDPCommand = command;
+            expectedWidth = metrics.width;
+            expectedHeight = metrics.height;
+            expectedDeviceScaleFactor = metrics.deviceScaleFactor;
+            expectedMobile = metrics.mobile;
+          }
+        });
+      };
+      client.api.setDeviceDimensions({width: 400, deviceScaleFactor: 100}, function (){
+        assert.strictEqual(expectedCDPCommand, 'Emulation.setDeviceMetricsOverride');
+        assert.strictEqual(expectedWidth, 400);
+        assert.strictEqual(expectedHeight, 0);
+        assert.strictEqual(expectedDeviceScaleFactor, 100);
+        assert.strictEqual(expectedMobile, false);
```

Comment:
I think it would be more prettier to create one object instead of multiple variables, if we can ignore negligible performance.

```
let expected;
client.transport.driver.createCDPConnection = function() {
        return Promise.resolve({
          execute: function(command, metrics) {
            expected = {
                 'CDPCommand': command;
                 'width': metrics.width;
                 'height': metrics.height;
                 'deviceScaleFactor': metrics.deviceScaleFactor;
                 'mobile': metrics.mobile;
             }
        });
      };
      client.api.setDeviceDimensions({width: 400, deviceScaleFactor: 100}, function (){
        assert.strictEqual(expected.CDPCommand, 'Emulation.setDeviceMetricsOverride');
        assert.strictEqual(expected.width, 400);
        assert.strictEqual(expected.height, 0);
        assert.strictEqual(expected.deviceScaleFactor, 100);
        assert.strictEqual(expected.mobile, false);
       });
      client.start(done);
    });
  });
```

Reply:
Yeah, I think that would be better. Will update the test code in all the PRs.

Reply:
Done.